### DISCUSSION TITLE
Looking for examples/ideas: multipage HTML documents with global index and search function

### DISCUSSION THREAD

**Discussion by mrestelli** [jgm/pandoc#10194] [Q&A]
Hi all,
   I am looking for examples on building multiple page HTML documents
from markdown input. The key requirement would be having a convenient
way to define links across different documents and having an index
linking all the pages.

Nice to have would be a global search function and a navigation bar,
similar to what mdbook does.

Thank you,
Marco

**Comment by jgm** [comment]
Have you explored pandoc's `chunkedhtml` output?

**Reply by mrestelli** [reply]
Very interesting indeed!

Based on some first tests:
- if I have many .md files, am I correct that I should pass all of them to pandoc at once, and they are processed as if they were just a single large file? In other words, the output is split according to headers inside the markdown, regardless of how it is distributed in multiple input files
- the same is true for filters: they see the whole text resulting from all input files
- I am not sure how to get a table of contents in each chunk: the manual says "To include it on every page, set the toc variable manually." How do I set this?

**Reply by jgm** [reply]
- Yes.
- Yes.
- `--variable toc`

**Reply by mrestelli** [reply]
OK, thnak you. Just few more questions:

1. when using `--variable toc` each page gets a complete TOC. If it
possible to modify this so that the TOC includes only the items below
the page level? So, for instance, the page corresponding to an H1
header should include all the H2 and higher headers in this section,
but nothing belonging to other H1 headers. More in general, is it
possible to customize the TOCs?

2. if I use
         ``` 
  --include-in-header=some_header.html
  --include-after-body=some_footer.html
         ``` 
the header and footer are included in each generated html file, which
is very convenient. Is there a way to adapt them to each output file,
for instance adding a "last modified" date different for each file?

**Reply by jgm** [reply]
1.  Not at this point. See #10130.
2. No, because the whole thing will be regenerated any time anything in the input changes.

**Reply by mrestelli** [reply]
OK, thank you. For 1., I have a working hack as follows: I use `--toc` to generate the general TOC, and I use a filter to include a page specific TOC after each H1 header (I am happy to share the filter if anybody is interested).

As for 2.: the "last modified" date was probably a bad example. I am looking for a general way to customize the footer for each HTML page. Ideally I would use a macro in `some_footer.html` like `LAST_MODIFIED_BY` and then define it on a per HTML page basis either from the command line or inside the markdown input. Is there any mechanism for doing this?

**Reply by jgm** [reply]
For including this in the footer, I don't see a good way.  The only template variable that is set differently depending on the chunk is going to be `pagetitle`.

**Reply by mrestelli** [reply]
I see. Maybe I can preprocess the md files and add the footer directly in the source.

This is probably not relevant any more for my question, but where is `pagetitle` set and how could I use it? More in general, I am confused about variables in pandoc: where are they set and where are they used?

**Reply by jgm** [reply]
Yes, you could add it in the source.
`pagetitle` in this case is computed from the section title and set automatically.
On variables in general, see the manual s.v. Variables.  `--variable` can set them on the command line, but they are also set from template variables, and some are set automatically by the writers.
